sensors/GoalSensor.py

```diff
@@ -0,0 +1,75 @@
+from sensors.SensorCommon import Sensor
+from sensors.SensorCommon import Dist2Circ
+import numpy as np
+import operator
+
+
+class GoalSensor(Sensor):
+
+    def __init__(self, nbGoals=0, SensorRange=10, mode="position"):
+        super().__init__(nbObservations=nbGoals, SensorRange=SensorRange)
+        self._Observation = np.ones([self._nbObservations, 2]) * self._limSensor
+        self._mode = mode
+        self._setSensorName()
+
+    def _setSensorName(self):
+        if self._mode == "position":
+            self._name = "GoalPosition"
+        elif self._mode == "distance":
+            self._name = "GoalDistance"
+
+    def name(self):
+        return self._name
+
+    def _reset(self):
+        self._Observation[:] = 0
+
+    def sense(self, s, goals, obstacles, t=0):
+        self._reset()
+        if self._mode == "position":
+            for idx, goal in enumerate(goals):
+                if idx >= self._nbObservations:
+                    break
+                self._Observation[idx][0] = goal.position(t=t)[0]
+                self._Observation[idx][1] = goal.position(t=t)[1]
+
+        elif self._mode == "distance":
+            for idx, goal in enumerate(goals):
+                if idx >= self._nbObservations:
+                    break
+                currGoalPos = goal.position(t=t)
+                currGoalDist = Dist2Circ(s['x'][0], s['x'][1], currGoalPos[0], currGoalPos[1], goal.epsilon())
```

Comment:
The variable `s` refers to the current state I assume. Maybe you can think of a more meaningful name here.
Then, you refer directly to the state as the forward kinematics of the point robot is the state essentially (configuration space equals workspace). Therefore, this does not work for other robots at the moment. 
We need to either pass the forward kinematics to the sensor on initialize forward kinematics in the sensor constructor. 
The second option seems better, as the goal object already contains information about which link the goal is defined for. We can discuss this later.

Reply:
Regarding the naming: Definetely!
kinematic chain: Exactly, right now it only works for the point robot. Would love to discuss how we can make this general for all robots!

Reply:
Let me know when you have it at a state that all the other issues and remarks are addressed. Then I will propose something to generalize it to generic kinematic chains.